dotnet/CoreLib/MemoryStorage/AzureCognitiveSearch/AzureCognitiveSearchMemory.cs

```diff
@@ -576,9 +576,13 @@ private static string BuildSearchFilter(ICollection<MemoryFilter> filters)
                 {
                     var fieldValue = keyValue.Value?.Replace("'", "''", StringComparison.Ordinal);
                     return $"tags/any(s: s eq '{keyValue.Key}{Constants.ReservedEqualsSymbol}{fieldValue}')";
-                });
+                })
+                .ToList();
 
-            conditions.Add($"({string.Join(" and ", filterConditions)})");
+            if (filterConditions is { Count: > 0 })
```

Comment:
how does the code produce an empty `filterConditions` list from a non empty `filters`?

Reply:
all calls to BuildSearchFilter are gated by a `if (filters is { Count: > 0 })` (line) so this function is always passed at least one filter. I can't see a scenario where `filterConditions` would be empty.

could you include a test to reproduce the error?

Reply:
This can happen if an empty instance of `MemoryFilter` is passed (see my comment [here](https://github.com/microsoft/kernel-memory/issues/109#issuecomment-1772800062)). The `string.Join` call assumes `filterConditions` is a non-empty collection which, when it is, just produces `()`. The current scenario is when setting `MemoryFilter.MinRelevance` without any filters being configured (see #109).

The scenario shouldn't really exist _but_ is still possible, so this change protects against it.

Reply:
is it possible to add a test to reproduce the error?